src/contexts/plotController/plotControllerContext.tsx

```diff
@@ -60,11 +61,15 @@ export function usePlotEvents(
 
   const ref = useRef(handlers);
   useEffect(() => {
-    ref.current = handlers;
-  }, [handlers]);
+    if (!disabled) {
+      ref.current = handlers;
+    }
+  }, [disabled, handlers]);
   useEffect(() => {
-    userContext.registerHandlers(ref);
-    return () => userContext.unregisterHandlers(ref);
+    if (userContext) {
+      userContext.registerHandlers(ref);
+      return () => userContext.unregisterHandlers(ref);
+    }
   }, [userContext]);
```

Comment:
I'm not sure it will work correctly if `disabled` changes dynamically. Can you add a story with a button to toggle the value of `disabled` ?